pkg/controller/miganalytic/restic_executor.go

```diff
@@ -0,0 +1,156 @@
+
+package miganalytic
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/konveyor/mig-controller/pkg/compat"
+	"github.com/konveyor/mig-controller/pkg/pods"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ResticPodLabelKey is the key of the label used to discover Restic pod
+	ResticPodLabelKey = "name"
+	// ResticPodLabelValue is the value of the label used to discover Restic pod
+	ResticPodLabelValue = "restic"
+)
+
+// ResticDFCommandExecutor uses Restic pods to run DF command
+type ResticDFCommandExecutor struct {
+	// Namespace is the ns in which Restic pods are present
+	Namespace string
+	// Client to interact with Restic pods
+	Client compat.Client
+	// ResticPodReferences is a local cache of known Restic pods
+	ResticPodReferences map[string]*corev1.Pod
+}
+
+// DF given a podRef and a list of volumes, runs df command, returns with structured command context
+// any errors running the df command are suppressed here. DFCommand.stdErr field should be used to determine failure
+func (r *ResticDFCommandExecutor) DF(podRef *corev1.Pod, persistentVolumes []MigAnalyticPersistentVolumeDetails) DFCommand {
+	// TODO: use the appropriate block size based on PVCs
+	dfCmd := DFCommand{
+		BaseLocation: "/host_pods",
+		BlockSize:    DecimalSIMega,
+		StdOut:       "",
+		StdErr:       "",
+	}
+	cmdString := dfCmd.PrepareDFCommand(persistentVolumes)
+	restCfg := r.Client.RestConfig()
+	podCommand := pods.PodCommand{
+		Pod:     podRef,
+		RestCfg: restCfg,
+		Args:    cmdString,
+	}
+	err := podCommand.Run()
+	if err != nil {
+		log.Info(
+			fmt.Sprintf("df command inside pod %s returned non-zero error code", podRef.Name))
+	}
+	dfCmd.StdErr = podCommand.Err.String()
+	dfCmd.StdOut = podCommand.Out.String()
+	return dfCmd
+}
+
+// getResticPodForNode lookup Restic Pod ref in local cache
+func (r *ResticDFCommandExecutor) getResticPodForNode(nodeName string) *corev1.Pod {
+	if podRef, exists := r.ResticPodReferences[nodeName]; exists {
+		return podRef
+	}
+	return nil
+}
+
+// loadResticPodReferences load Restic Pod refs in-memory
+func (r *ResticDFCommandExecutor) loadResticPodReferences() error {
+	if r.ResticPodReferences == nil {
+		r.ResticPodReferences = make(map[string]*corev1.Pod)
+	}
+	resticPodList := corev1.PodList{}
+	labelSelector := client.InNamespace(r.Namespace).MatchingLabels(
+		map[string]string{
+			ResticPodLabelKey: ResticPodLabelValue})
+	err := r.Client.List(context.TODO(), labelSelector, &resticPodList)
+	if err != nil {
+		return err
+	}
+	for i := range resticPodList.Items {
+		if resticPodList.Items[i].Spec.NodeName != "" {
+			r.ResticPodReferences[resticPodList.Items[i].Spec.NodeName] = &resticPodList.Items[i]
+		}
+	}
+	return nil
+}
+
+// Execute given a map node->[]pvc, runs Df command for each, returns list of structured df output per pvc
+func (r *ResticDFCommandExecutor) Execute(pvcNodeMap map[string][]MigAnalyticPersistentVolumeDetails) ([]DFOutput, error) {
```

Comment:
I wonder if we should consider bumping `MaxConcurrentReconciles` to `10` instead of `1` so that the PV resize MigAnalytic is never stuck waiting in line behind the general purpose MigAnalytic that pulls all resource counts for plan.

https://github.com/konveyor/mig-controller/blob/master/pkg/controller/miganalytic/miganalytics_controller.go#L79